### PR TITLE
Update release.yaml to fetch changelog

### PR DESCRIPTION
**Pull request by elliotmjackson**
Small bash step to fetch changelog of current release version from changelog before putting it into the body of the release.

[example](https://github.com/elliotmjackson/buf/releases/tag/v1.12.0)

### CODE REVIEW

**Comment by bufdev** [comment]
Just to make sure this is covered - this does not fix the core issue. With this change, the release notes that will go out to watchers will now just be empty, instead of having the release notes that we want.

**Comment by elliotmjackson** [comment]
yes, correct. Thats coming up next

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -144,12 +144,14 @@ jobs:
         run: |
           unset RELEASE_MINISIGN_PRIVATE_KEY
           unset RELEASE_MINISIGN_PRIVATE_KEY_PASSWORD
+      - name: Fetch Changelog
+        run: .github/actions/changelog-action/entrypoint.sh > ${{env.VERSION}}-CHANGELOG.md
```
Comment (bufdev) [inline .github/workflows/release.yaml:148]: There is no chance this gets published anywhere somehow correct? Ie we don't need to put this in some special temp directory?

Comment (elliotmjackson) [reply .github/workflows/release.yaml:148]: the temp directory is the action itself which is crushed when it terminates

File: .github/actions/changelog-action/entrypoint.sh [hunk 2]
```diff
@@ -0,0 +1,19 @@
+#!/bin/bash
```
[the remark this answers is not in the document]
Comment (elliotmjackson) [reply .github/actions/changelog-action/entrypoint.sh:1]: 😆 i think i managed to not conform with every part of the style guide

File: .github/workflows/release.yaml (hunk 1 above)
Comment (elliotmjackson) [reply .github/workflows/release.yaml:148]: I could delete the file in a subsequent test but i think its unnecessary

Comment (bufdev) [reply .github/workflows/release.yaml:148]: Definitely unnecessary, just making sure

File: .github/actions/changelog-action/entrypoint.sh (hunk 2 above)
[the remark this answers is not in the document]
Comment (bufdev) [reply .github/actions/changelog-action/entrypoint.sh]: Well you're a champion of confirming to any style guide once you find it, it looks perfect now :-)

**Review by bufdev — APPROVED** [review]
Thanks for putting this together

**Pull request merged**